src/main/scala/strawman/collection/SortedOps.scala

```diff
@@ -3,10 +3,12 @@ package strawman.collection
 import scala.{Ordering, Option, Some}
 
 /** Base trait for sorted collections */
-trait SortedOps[A, +C] {
+trait SortedOps[A, +C, +CC[_]] {
 
   implicit def ordering: Ordering[A]
 
+  def sortedIterableFactory: SortedIterableFactory[CC]
```

Comment:
This PR also adds the another factor kinds. So far every collection exposes an `IterableFactory`. Now in addition to that every sorted collection has a `SortedIterableFactory`. We'll probably want to add `MapFactory` and `SortedMapFactory` types to the appropriate collection types as well.